src/main/java/io/gingersnapproject/cdc/cache/hotrod/HotRodService.java

```diff
@@ -22,86 +34,87 @@
 import org.infinispan.commons.configuration.StringConfiguration;
 import org.infinispan.commons.dataconversion.MediaType;
 import org.infinispan.commons.util.concurrent.CompletableFutures;
-
-import javax.annotation.PostConstruct;
-import javax.enterprise.context.ApplicationScoped;
-import javax.enterprise.event.Observes;
-import javax.inject.Inject;
-
-import java.util.Map;
-import java.util.concurrent.CompletableFuture;
-import java.util.concurrent.CompletionStage;
-import java.util.concurrent.ConcurrentHashMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 @ApplicationScoped
 public class HotRodService implements CacheService {
+   private static final Logger log = LoggerFactory.getLogger(HotRodService.class);
    private static final String OFFSET_CACHE_NAME = "___debezium-offset";
    private static final String SCHEMA_CACHE_NAME = "___debezium-schema";
 
    @Inject NotificationManager eventing;
 
-   @Inject Configuration config;
-
    @Inject DBSyncerMetrics metrics;
 
-   volatile RemoteCacheManager rcm;
-
-   private final Map<String, CacheBackend> backends = new ConcurrentHashMap<>();
+   private final Map<URI, HotRodCache> managers = new ConcurrentHashMap<>();
    private volatile boolean shutdownSignal;
```

Comment:
I don't see the point of this field any longer. It seems like it is there now to allow for a concurrent stop of a cache to help out with shutdown? I think this and the call to `tryDestroy` in the stop method should be removed.

Reply:
Yeah, originally was because of multiple places calling shutdown. It should be good now, only calling `stop` in a callback invoked once the Debezium engine stops. It also avoids the issue of stopping the cache manager before flushing the pending work.